src/Psalm/Internal/Analyzer/Statements/Expression/Call/FunctionCallAnalyzer.php

```diff
@@ -1003,7 +1003,20 @@ private static function checkFunctionCallPurity(
                 && !$context->inside_conditional
                 && !$context->inside_unset
             ) {
-                if (!$context->inside_assignment && !$context->inside_call && !$context->inside_use) {
+                /**
+                 * If a function is pure, and has the return type of 'no-return',
+                 * it's okay to dismiss it's return value.
+                 */
+                if (
+                    !$context->inside_assignment &&
+                    !$context->inside_call &&
+                    !$context->inside_use &&
+                    !(
+                        $function_call_info->function_storage &&
+                        $function_call_info->function_storage->return_type &&
+                        (array_values($function_call_info->function_storage->return_type->getAtomicTypes())[0] ?? null) instanceof Type\Atomic\TNever
```

Comment:
I replaced this with `$function_call_info->function_storage->return_type->isNever()`

Reply:
i didn't notice the method existed 😅